kernel/src/snapshot.rs

```diff
@@ -152,12 +152,32 @@ impl Snapshot {
                 (Some(cp), None) => {
                     list_log_files_with_checkpoint(&cp, fs_client.as_ref(), &log_url)?
                 }
-                (Some(cp), Some(version)) if cp.version >= version => {
+
+                (Some(cp), Some(version)) if cp.version <= version => {
+                    list_log_files_with_checkpoint(&cp, fs_client.as_ref(), &log_url)?
+                }
+
+                (Some(cp), Some(version)) if cp.version > version => {
```

Comment:
I  feel like I might be missing something, but it seems like we're doing the same thing if `cp.version <= version` or if `cp.version > version`. That is, we call `list_log_files_with_checkpoint` in either case.

So... do we need two case statements, or can we just remove the cp.version and version comparison and have one case?